logic/standalone/waterfall.py

```diff
@@ -0,0 +1,129 @@
+'''
+Logic module that can create waterfall objects.
+
+
+USAGE EXAMPLE:
+
+'''
+
+import logic.common.log_utils as log
+import logic.common.tiled_utils as tiled_utils
+
+#--------------------------------------------------#
+'''Variables'''
+
+TILELAYER_PREFIX = "_falls"
+
+DEFAULT_SORT_FG = "fg_tiles,15"
+DEFAULT_SORT_BG = "bg_tiles,5"
+DEFAULT_SORT = DEFAULT_SORT_BG
+
+DEFAULT_COLOR = "ffffff"
+
+OBJECTGROUP_PREFIX = "objects_waterfall_"
+
+
+
+#--------------------------------------------------#
+'''Public Functions'''
+
+def GetMetaDataFromBlockout(playdo):
+	'''Obtain the tile2d and various meta data for the waterfall, based on tilelayer name
+	  e.g. _falls => sort_value is DEFAULT_SORT, color is DEFAULT_COLOR
+	       _falls_fg15 => sort_value is "fg_tiles,15", color is DEFAULT_COLOR
+	       _falls_bg_bbccff => sort_value is DEFAULT_SORT_BG, color is "#bbccff/1f"
```

Comment:
(continued from above) the usage would NOT be like this.

The goal is to not have to set the sort layers or color manually. Still, it's good we did this since I was able to get into your headspace and see what you were thinking. I think the best way to convey what the tool needs to do is to get you an example input level XML. So I'll send that over soon.